GUI/src/javafxgui/JavaFXComponents/GUIView.java

```diff
@@ -20,14 +19,10 @@ public GUIView(Scene scene) {
         this.controller = null;
     }
 
-    public void setController(Controller controller) {
+    public void setController(GUIAppController controller) {
```

Comment:
Can you inject the dependency via the constructor instead? I dislike setters like this for DI.